### PR TITLE
[feature] fetch all replies

### PR DESCRIPTION
**Pull request by sneakers-the-rat**
Adds a button to the bottom of an expanded status to fetch all replies from the OP server (or as many as they will give us).

- Example: https://neuromatch.social/@jonny/109582537808504261
- Context: https://neuromatch.social/tags/FetchAllReplies
- Wiki: https://wiki.neuromatch.io/Fetch_All_Replies

The feature:
- Tries to figure out the API URL for the OP server (currently just using the masto pattern, consider that a TODO)
- Fetches the replies from the /statuses/{:id}/context endpoint
- Uses the search API on the host instance in order to fetch the necessary status and account information to display the status.

It also:
- Displays a dialogue box showing how many statuses it is grabbing

Still to do:
- Write tests (dont know hwo to do that in ruby)
- Catch errors better
- Insert into thread in correct order (refreshing fixes)

### CODE REVIEW

**Review by mannazsci — COMMENTED** [review]
minor comments on documentation, status visibility, non-mastodon domain handling, and accessibility.

File: app/javascript/flavours/glitch/actions/statuses.js [hunk 1]
```diff
@@ -219,6 +225,65 @@ export function fetchContextFail(id, error) {
...
+  // since I don't know a better way of finding it
+  let { origin, pathname } = new URL(status_url);
+  let paths = pathname.split('/');
+  let external_id = paths[paths.length - 1];
+
+  return (dispatch, getState) => {
+    dispatch(fetchExternalContextRequest(id));
+
+    api(getState).get(`${origin}/api/v1/statuses/${external_id}/context`).then(response => {
+      dispatch(showAlert('Getting Replies...', `Getting ${response.data.descendants.length} Replies.`));
```
[the remark this answers is not in the document]
Comment (sneakers-the-rat) [reply app/javascript/flavours/glitch/actions/statuses.js:242]: good point, I gotta see how the translation system works and then I'll change

File: app/javascript/flavours/glitch/actions/statuses.js [hunk 2]
```diff
@@ -219,6 +225,65 @@ export function fetchContextFail(id, error) {
...
 
+
+
+export function fetchExternalContext(status_url, id){
+  // Get just the protocol and domain of the instance
+  // note that we are just assuming that the api is at <domain>/api
+  // since I don't know a better way of finding it
+  let { origin, pathname } = new URL(status_url);
+  let paths = pathname.split('/');
+  let external_id = paths[paths.length - 1];
```
[the remark this answers is not in the document]
Comment (sneakers-the-rat) [reply app/javascript/flavours/glitch/actions/statuses.js:236]: no i don't think so, each instance makes its own local IDs for posts that it uses internally which don't necessarily match the Ids used by other instances.

**Comment by sneakers-the-rat** [comment]
Closing in favor of: https://github.com/NeuromatchAcademy/mastodon/pull/44

**Pull request closed without merging**